File: scratch/extract_from_cdp.py
import json
import os
import sys
from playwright.sync_api import sync_playwright

if hasattr(sys.stdout, 'reconfigure'):
    sys.stdout.reconfigure(encoding='utf-8')

def extract_from_cdp():
    print("Connecting to Chrome on port 9222 via CDP...")
    with sync_playwright() as p:
        try:
            browser = p.chromium.connect_over_cdp("http://127.0.0.1:9222")
        except Exception as e:
            print(f"Failed to connect to CDP on 9222: {e}")
            return

        contexts = browser.contexts
        print(f"Found {len(contexts)} contexts.")
        target_page = None

        for ctx in contexts:
            for page in ctx.pages:
                url = page.url
                print(f"Open tab: {url}")
                if "dbdbschool.kr" in url and "modifyField" in url:
                    target_page = page
                    break
            if target_page:
                break

        if not target_page:
            for ctx in contexts:
                for page in ctx.pages:
                    if "dbdbschool.kr" in page.url:
                        target_page = page
                        print(f"Found dbdbschool tab, navigating to modifyField...")
                        target_page.goto("https://www.dbdbschool.kr/af/ad_lec/modifyField/p/1/sn/3267/sld/11/sof/ln/sot/asc", wait_until="domcontentloaded")
                        break
                if target_page:
                    break

        if not target_page:
            print("No dbdbschool tab found!")
            return

        print(f"Target page URL: {target_page.url}")
        print(f"Target page Title: {target_page.title()}")

        styles = target_page.evaluate("""
            () => Array.from(document.querySelectorAll('link[rel="stylesheet"]')).map(l => l.outerHTML).join('\\n')
        """)

        body_html = target_page.evaluate("""
            () => {
                const el = document.querySelector('#contents') || document.querySelector('#content') || document.querySelector('form[name="fm_write"]') || document.body;
                return el.outerHTML;
            }
        """)

        full_content = f"<!-- Auto-extracted via CDP Port 9222 -->\\n<!-- URL: {target_page.url} -->\\n<!-- Title: {target_page.title()} -->\\n\\n<!-- Stylesheets -->\\n{styles}\\n\\n<!-- Extracted DOM -->\\n{body_html}"

        with open("scratch/target.html", "w", encoding="utf-8") as f:
            f.write(full_content)

        print(f"✅ Successfully wrote scratch/target.html ({len(full_content)} chars)!")

        # Save cookies to auth.json
        cookies = target_page.context.cookies()
        auth_state = {
            "cookies": cookies,
            "origins": [
                {
                    "origin": "https://www.dbdbschool.kr",
                    "localStorage": []
                }
            ]
        }
        with open("auth.json", "w", encoding="utf-8") as f:
            json.dump(auth_state, f, indent=2)
        print(f"✅ Saved fresh auth.json with {len(cookies)} cookies!")

        target_page.screenshot(path="scratch/target.png")
        print("✅ Saved scratch/target.png!")

if __name__ == "__main__":
    extract_from_cdp()

File: scratch/extract_from_cdp.js
const { chromium } = require('playwright');
const fs = require('fs');
const path = require('path');

async function extractFromLiveChrome() {
    console.log('Connecting to Chrome on port 9222...');
    const browser = await chromium.connectOverCDP('http://127.0.0.1:9222');
    const contexts = browser.contexts();
    console.log('Contexts found:', contexts.length);

    let targetPage = null;
    for (const ctx of contexts) {
        const pages = ctx.pages();
        for (const p of pages) {
            const url = p.url();
            console.log('Open tab URL:', url);
            if (url.includes('dbdbschool.kr') && url.includes('modifyField')) {
                targetPage = p;
                break;
            }
        }
        if (targetPage) break;
    }

    if (!targetPage) {
        console.log('modifyField page not found directly, looking for any dbdbschool tab...');
        for (const ctx of contexts) {
            for (const p of ctx.pages()) {
                if (p.url().includes('dbdbschool.kr')) {
                    targetPage = p;
                    console.log('Found dbdbschool tab, navigating to modifyField...');
                    await targetPage.goto('https://www.dbdbschool.kr/af/ad_lec/modifyField/p/1/sn/3267/sld/11/sof/ln/sot/asc', { waitUntil: 'domcontentloaded' });
                    break;
                }
            }
            if (targetPage) break;
        }
    }

    if (!targetPage) {
        console.error('No dbdbschool tab found in Chrome!');
        await browser.close();
        process.exit(1);
    }

    console.log('Extracting DOM from target page:', targetPage.url());
    const data = await targetPage.evaluate(() => {
        const styles = Array.from(document.querySelectorAll('link[rel="stylesheet"]')).map(l => l.outerHTML).join('\n');
        const contentsEl = document.querySelector('#contents') || document.querySelector('#content') || document.querySelector('form[name="fm_write"]') || document.body;
        return {
            styles,
            fullHtml: document.documentElement.outerHTML,
            bodyHtml: contentsEl ? contentsEl.outerHTML : document.body.outerHTML,
            url: window.location.href,
            title: document.title
        };
    });

    const targetHtmlPath = path.resolve('scratch/target.html');
    const fullContent = `<!-- Auto-extracted via CDP Port 9222 -->\n<!-- URL: ${data.url} -->\n<!-- Title: ${data.title} -->\n<!-- Extracted Time: ${new Date().toISOString()} -->\n\n<!-- Stylesheets -->\n${data.styles}\n\n<!-- Extracted DOM -->\n${data.bodyHtml}`;

    fs.writeFileSync(targetHtmlPath, fullContent, 'utf-8');
    console.log(`Successfully saved scratch/target.html (${fullContent.length} bytes)!`);

    // Also update auth.json with fresh storage_state
    const cookies = await targetPage.context().cookies();
    const authState = {
        cookies: cookies,
        origins: [
            {
                origin: "https://www.dbdbschool.kr",
                localStorage: []
            }
        ]
    };
    fs.writeFileSync('auth.json', JSON.stringify(authState, null, 2), 'utf-8');
    console.log(`Saved fresh auth.json with ${cookies.length} cookies!`);

    await targetPage.screenshot({ path: 'scratch/target.png' });
    console.log('Saved scratch/target.png!');
}

extractFromLiveChrome().catch(err => {
    console.error('Error:', err);
    process.exit(1);
});
